### PR TITLE
[RLlib] Fix FeatureImportance OfflineEvaluator construction.

### PR DESCRIPTION
**Pull request by gjoliver**
Signed-off-by: Jun Gong <jungong@anyscale.com>

## Why are these changes needed?

FeatureImportance offline evaluator is currently broken.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -680,6 +681,7 @@ def setup(self, config: PartialAlgorithmConfigDict):
                 mod, obj = method_type.rsplit(".", 1)
                 mod = importlib.import_module(mod)
                 method_type = getattr(mod, obj)
+
             if isinstance(method_type, type) and issubclass(
```
Comment (kouroshHakha) [inline rllib/algorithms/algorithm.py:685]: `OffPolicyEstimator` is a subclass of `OfflineEvaluator`. So I think the code between these two conditions can be shared. You basically need to instantiate the classes in the end but the only difference is to add `model_config['gamma'] = self.config["gamma"]` only if it's also a subclass of `OffPolicyEstimator`.

**Review by kouroshHakha — APPROVED** [review]

**Review by kouroshHakha — COMMENTED** [review]
Quick question. Do we not have an integration test for FeatureImportance? should we add it in this PR so this should have been caught?

File: rllib/algorithms/algorithm.py (hunk 1 above)
Comment (gjoliver) [reply rllib/algorithms/algorithm.py:685]: ok, updated. 👍

**Comment by gjoliver** [comment]
> Quick question. Do we not have an integration test for FeatureImportance? should we add it in this PR so this should have been caught?

right. I added a TODO for you to add the test after you come back man.
gonna keep this PR small, so I can cherry pick it into the release branch.

**Pull request merged**